### ISSUE TITLE
Prevent sync when local data is corrupted

### ISSUE THREAD

**Issue by TheSlimvReal** [Aam-Digital/ndb-core#815]
The CouchDB backups can be used to reset the state of the database.
However, each user also has a local copy of the database in the browser.
If the backup is now used to delete some (wrongly added) documents, but some users have these wrong documents in their local database, then the backup will be overridden once these users synchronize their local database with the remote one.

To prevent this the app automatically has to detect if a backup has been introduced and when a user has last synchronized with the remote database.
In case the database is backed up to a state **older** than the last synchronization of a user, this users local database should be deleted and not synced with the remote one.
Otherwise the user does not have the corrupted data locally and can safely synchronize with the backup of the remote database.

To achieve this, the following steps have to be taken:

- Save the date of the last synchronization in the local storage
- When introducing a backup, save the date to which the system is reset in  the `config.json` (e.g. `backup: 11.05.2021`)
- Before the app synchronizes with a remote database, check if the `AppConfig` holds a backup date **older** than the last sync. If this is the case, remove the local (in-browser) database before starting the sync with the remote database
- After deleting the local database, save backup date in the local storage
- When the app next time checks if a backup happened, it should not delete the local database if the backup date is equal to the one in the local storage

In short:

- [ ] Save date of last sync in local storage
- [ ] Save date of last backup in local storage
- [ ] Check for a new backup before syncing
- [ ] Delete local storage if it conflicts with the backup
